src/js/view/controls/controls.js

```diff
@@ -408,6 +409,11 @@ export default class Controls {
                 timeout || ACTIVE_TIMEOUT);
         }
         if (!this.showing) {
+            const mute = this.mute;
+
+            if (mute && mute.element().style.display === '') {
+                utils.removeClass(this.playerContainer, 'jw-flag-autostart');
+            }
```

Comment:
Implementing logic based on DOM state is not a good practice - if we change how mute is displayed this breaks.

What's the thinking here?

There is existing code ("autostartMuted" logic) that I would expect to have removed or changed as part of this improvement, instead this PR just adds more code on top.

Reply:
@egreaves can you provide an overview of the code you added for the initial implementation (what these props are and if they are needed given the scope of the changes)?

I would like us to remove these if they are no longer needed:

- model "autostartMuted"
- CSS .jw-flag-autostart
- button .jw-autostart-mute